### DISCUSSION TITLE
Auto Submit for Previous Weeks?

### DISCUSSION THREAD

**Discussion by KatjaGlassConsulting** [KatjaGlassConsulting/ApprovalBundle#39] [Ideas]
@corrigenda-admin brought up the idea to Auto-Submit for previous weeks (#38).

> This is more a feature request, currently when someone tries to Submit a week but missed a week it comes up with the warning:
> 
> "Please add previous weeks to approve".
> 
> When people have a couple of weeks off on holiday, they have to go back and do previous weeks. Is it possible to get it to auto submit any previous weeks by default? Like a round up? Wasnt sure if thats possible?

I think such a functionality might quite often be ticked also accidently. I would recommend to always check-in every week for approval after review, otherwise a week might be accidently submitted without even checking. Personally I think the risk is higher than the benefit. For this I closed the issue, but everyone can discuss here. When more find such a thing useful, it might become implemented.

**Comment by duyanhtruong** [comment]
Hello,
This became a problem for my specific use case. I imported massive amounts of timesheet entries (20 years worth of old timesheet records), and now it won't let me approve any recent timesheet. I would have to go through and manually approve all weekly entries starting from 20 years ago to be able to approve anything recent.

A feature to automatically submit and approve all past timesheets would be of great help. 

> I think such a functionality might quite often be ticked also accidently. I would recommend to always check-in every week for approval after review, otherwise a week might be accidently submitted without even checking. Personally I think the risk is higher than the benefit.

I understand the risk you mentioned, so I was wondering if we could make auto-submit and auto-approve a feature but locate it at a very specific place or setting so that it cannot be "accidentally" used?

Or we could create an checkbox option that allows timesheets to be approved without having to approve the previous weeks' timesheets?

**Reply by KatjaGlassConsulting** [reply]
@duyanhtruong ,

In the settings tab there is an approval start date. When you set it to this week, then only starting with this week, this and later weeks need approval.

**Reply by duyanhtruong** [reply]
I see. That did it. Thank you and sorry I didn't find that before; I'm new to Kimai.
On a side question, I was wondering when/if the Approval bundle will support approving weeks starting on Sundays?

**Reply by KatjaGlassConsulting** [reply]
Hi, the Sunday support is not planned as this is not trivial and very time consuming to implement.
